registry/registry.go

```diff
@@ -150,3 +153,42 @@ func (z *zookeeperRegistry) deleteNode(path string) error {
 	}
 	return nil
 }
+
+func (z *zookeeperRegistry) keepalive(path string, content []byte) {
+	sessionID := z.conn.SessionID()
+	ticker := time.NewTicker(z.sessionTimeout / 2)
+	defer ticker.Stop()
+	for range ticker.C {
+		cur := z.conn.SessionID()
+		if cur > 0 && sessionID != cur {
+			if err := z.ensureName(path, content, zk.FlagEphemeral); err == nil {
+				sessionID = cur
+			}
+		}
+	}
+}
+
+func (z *zookeeperRegistry) ensureName(path string, data []byte, flags int32) error {
+	exists, stat, err := z.conn.Exists(path)
+	if err != nil {
+		return err
+	}
+	if exists && flags&zk.FlagEphemeral == zk.FlagEphemeral {
```

Comment:
If the node still exists, return directly. no need delete and create/

Reply:
I want to fix a race condition if the server carshes after it creates the node without using CreateProtectedEphemeralSequential()

https://github.com/go-zookeeper/zk/blob/v1.0.3/conn.go#L1102

Reply:
Fixed.